engine/language-server/src/main/scala/org/enso/languageserver/runtime/ApiEventsMonitor.scala

```diff
@@ -0,0 +1,77 @@
+package org.enso.languageserver.runtime
+
+import org.enso.languageserver.monitoring.EventsMonitor
+import org.enso.polyglot.runtime.Runtime.ApiEnvelope
+import org.enso.polyglot.runtime.Runtime.Api
+
+import java.nio.charset.StandardCharsets
+import java.nio.file.{Files, Path, StandardOpenOption}
+import java.time.Clock
+
+/** Gather messages between the language server and the runtime and write them
+  * to the provided file in CSV format.
+  *
+  * @param path the path where to write the events
+  * @param clock the system clock
+  */
+final class ApiEventsMonitor(path: Path, clock: Clock) extends EventsMonitor {
+
+  import ApiEventsMonitor.Direction
+
+  /** @inheritdoc */
+  override def registerEvent(event: Any): Unit =
+    event match {
+      case envelope: ApiEnvelope =>
+        registerApiEvent(envelope)
+      case RuntimeConnector.MessageFromRuntime(envelope) =>
+        registerApiEvent(envelope)
+      case _ =>
+    }
+
+  def registerApiEvent(event: ApiEnvelope): Unit =
+    event match {
+      case Api.Request(requestId, payload) =>
+        Files.write(
```

Comment:
Wouldn't it be useful to start `Sampler` (as provided by [3389](https://github.com/enso-org/enso/pull/3389)) at this point and start collecting data?